rust/operator-binary/src/airflow_controller.rs

```diff
@@ -1079,15 +1059,27 @@ fn build_server_rolegroup_statefulset(
     }
 
     if merged_airflow_config.logging.enable_vector_agent {
-        pb.add_container(build_logging_container(
-            resolved_product_image,
-            merged_airflow_config
-                .logging
-                .containers
-                .get(&Container::Vector),
-        )?);
+        match airflow
+            .spec
+            .cluster_config
+            .vector_aggregator_config_map_name
+            .to_owned()
+        {
+            Some(vector_aggregator_config_map_name) => {
+                pb.add_container(build_logging_container(
+                    resolved_product_image,
+                    merged_airflow_config
+                        .logging
+                        .containers
+                        .get(&Container::Vector),
+                    &vector_aggregator_config_map_name,
```

Comment:
With the above suggestion, this would change to:

```suggestion
                    vector_aggregator_config_map_name,
```